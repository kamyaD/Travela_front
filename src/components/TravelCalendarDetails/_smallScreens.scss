@media screen and (min-width: 320px) and (max-width: 480px){
.user {
  .toggle-btn {
    top: 15px;
    right: 7px;
  }
}

.flight-details {
  overflow-x: scroll;

  .flight {
    width: 100%;
  }
}
}
