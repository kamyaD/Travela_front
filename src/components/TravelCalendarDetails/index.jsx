import React, {PureComponent, Fragment} from 'react';
import { PropTypes } from 'prop-types';

import './index.scss';

import ImageLink from '../image-link/ImageLink';
import upic from '../../images/upic.svg';
import upArrow from '../../images/upArrow.svg';
import downArrow from '../../images/downArrow.svg';
import flightTakeoff from '../../images/flightTakeoff.svg';
import flightLand from '../../images/flightLand.svg';

class TravelCalendarDetails extends PureComponent{
  constructor(props) {
    super(props);

    this.state = {
      hideDetails: true,
      hideBtnLabel: 'Show Details'
    };
  }
  renderUserBio= (name, role, department) => {
    const {hideDetails, hideBtnLabel} = this.state;
    return (
      <Fragment>
        <div className="user">
          <ImageLink
            imageSrc={upic}
            altText="Andela Logo"
            imageClass="navbar__mdl-upic"
          />
          <span className="user__bio">
            <p className="user__name">{name}</p>
            <span className="user__role">{role}</span>
            <span className="user__role">, </span>
            <span className="user__role">{department}</span>
          </span>
          <button className="toggle-btn" type="button" onClick={this.handleHideToggle}>
            {hideBtnLabel}
            <img className="toggle-btn__icon" src={hideDetails?downArrow:upArrow} alt="show/hide details" />
          </button>
        </div>
      </Fragment>
    );
  }

  renderFlightDetails=(flightType, flightDetails)=>{
    const headerAttrs = ['Airline', 'Flight No.', 'Arrival Time', 'Destination'];
    const dataAttrs = ['airline', 'flight_no', 'arrival_time', 'destination'];
    const{hideDetails} = this.state;
    const icon = flightType==='Arrival'?flightLand: flightTakeoff;

    if(!hideDetails){return(
      <Fragment>
        <div className="flight">
          <div className="flight__header">
            <img src={icon} alt="Arrival" />
            <p className="flight__title">{flightType}</p>
          </div>
          <table>
            {headerAttrs.map(attr => (<th key={attr}>{attr}</th>))}
            <tr>
              {dataAttrs.map(attr => (<td key={attr}>{flightDetails[attr]}</td>))}
            </tr>
          </table>
        </div>
      </Fragment>
    );}
  }

  renderTravelCalendarDetails= () => {
    const {calendar} = this.props;
    if(calendar){
      const {flights, name, role, department} = calendar;
      const {arrival, depature} = flights;

      return (
        <Fragment>
          <div className="demo-card-wide mdl-card mdl-shadow--2dp">
            {this.renderUserBio(name, role, department)}
            <div className="flight-details">
              {this.renderFlightDetails('Arrival', arrival)}
              {this.renderFlightDetails('Departure', depature)}
            </div>
          </div>
        </Fragment>
      );
    }
  }

  handleHideToggle = () => {
    const {hideDetails} = this.state;
    hideDetails ? this.setState({ hideDetails: false, hideBtnLabel:'Hide Details'})
      : this.setState({ hideDetails: true, hideBtnLabel:'Show Details'});
  }

  render(){
    return (
      <Fragment>
        <div>
          {this.renderTravelCalendarDetails()}
        </div>
      </Fragment>
    );
  }
}

TravelCalendarDetails.propTypes = {
  calendar: PropTypes.object.isRequired
};

export default TravelCalendarDetails;
