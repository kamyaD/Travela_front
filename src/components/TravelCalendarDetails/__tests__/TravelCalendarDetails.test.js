import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import TravelCalendarDetails from '../index';

describe('Travel Calendar Details', () => {
  const calendar = {
    "name": "Harrison Maina",
    "department": "Apprenticeship Department",
    "role": "Product designer",
    "flights": {
      "arrival": {
        "destination": "Lagos",
        "airline": "Kenya Airways",
        "flight_no": "KQ5752",
        "arrival_time": "24/10/2018 13:23:33",
        "depature_time": "24/10/2018 13:23:33"
      },
      "depature": {
        "destination": "Nairobi",
        "airline": "Kenya Airways",
        "flight_no": "KQ5752",
        "arrival_time": "24/10/2018 13:23:33",
        "depature_time": "24/10/2018 13:23:33"
      }
    }
  }

  it('should render TravelCalendarDetails', ()=>{
    const wrapper = shallow(<TravelCalendarDetails calendar={calendar}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show/hide details when button is clicked', () => {
    const wrapper = shallow(<TravelCalendarDetails calendar={calendar}/>);
    wrapper.find('button').simulate('click');
    expect(wrapper.find('button').text()).toEqual('Hide Details');
    wrapper.find('button').simulate('click');
    expect(wrapper.find('button').text()).toEqual('Show Details');
  });
});