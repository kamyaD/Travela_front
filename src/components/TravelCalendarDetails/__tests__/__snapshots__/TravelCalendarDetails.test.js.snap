// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Travel Calendar Details should render TravelCalendarDetails 1`] = `
<Fragment>
  <div>
    <div
      className="demo-card-wide mdl-card mdl-shadow--2dp"
    >
      <div
        className="user"
      >
        <ImageLink
          altText="Andela Logo"
          imageClass="navbar__mdl-upic"
          imageSrc="upic.svg"
        />
        <span
          className="user__bio"
        >
          <p
            className="user__name"
          >
            Harrison Maina
          </p>
          <span
            className="user__role"
          >
            Product designer
          </span>
          <span
            className="user__role"
          >
            , 
          </span>
          <span
            className="user__role"
          >
            Apprenticeship Department
          </span>
        </span>
        <button
          className="toggle-btn"
          onClick={[Function]}
          type="button"
        >
          Show Details
          <img
            alt="show/hide details"
            className="toggle-btn__icon"
            src="downArrow.svg"
          />
        </button>
      </div>
      <div
        className="flight-details"
      />
    </div>
  </div>
</Fragment>
`;
